src/Cryptol/Eval/What4.hs

```diff
@@ -177,6 +177,7 @@ suiteBPrims :: W4.IsSymExprBuilder sym => What4 sym -> Map.Map PrimIdent (Prim (
 suiteBPrims sym = Map.fromList $ [ (suiteBPrim n, v) | (n,v) <- prims ]
  where
  (~>) = (,)
+ vSeq n w = mkSeq sym (Nat n) (TVSeq w TVBit) 
```

Comment:
There are several parts of the code where we redefine a helper function like this one. Perhaps it would be worth giving this code a home in `Cryptol.Eval.Value` and exporting it?

Reply:
Is there a good way to indicate the role of each Integer parameter? With the definition nearby it's clear what role each plays, but pulling it out into `Cryptol.Eval.Value` it's a bit obscure to see a function like `wordSeq :: Backend sym => sym -> Integer -> Integer -> SEval sym (Genvalue sym)`.

Reply:
I would advise to just add Haddock comments for each `Integer`:

```hs
wordSeq ::
  Backend sym =>
  sym ->
  -- | The length of the sequence.
  Integer ->
  -- | The word size of the element type.
  Integer ->
  ...
```

Reply:
Alternatives to just Haddocks to indicate the usage of the arguments are: use a `newtype` wrapper around the `Integer`, or use the `tagged` package (https://hackage.haskell.org/package/tagged).  Both have some (small) impact on clients... which could be considered a good thing.

Reply:
True, although I suspect that there are many other places where we'd want to use such a newtype beyond `wordSeq`.

For now, I'm inclined to keep things simple and use two `Integer`s as arguments. If we encounter bugs later that are caused by confusing these two arguments, then we can consider introducing newtype(s) to distinguish them.